code/modules/mob/living/simple_animal/eldritch_demons.dm

```diff
@@ -0,0 +1,276 @@
+			current.icon_living = "armsy_mid"
+			current.front = src
+			current.AIStatus = AI_OFF
+			back = current
+		if(i > 0 && i < len)
+			current = new type(drop_location(),spawn_more = FALSE)
+			prev.back = current
+			prev.icon_state = "armsy_mid"
+			prev.icon_living = "armsy_mid"
+			prev.front = next
+			prev.AIStatus = AI_OFF
+		if(i == len)
+			prev.icon_state = "armsy_end"
+			prev.icon_living = "armsy_end"
+			prev.front = next
+			prev.AIStatus = AI_OFF
+		next = prev
+
+/mob/living/simple_animal/hostile/eldritch/armsy/Moved()
+
+	if(pulledby && !allow_pulling)
+		pulledby.stop_pulling()
+		loc = oldloc
+
+	if(back && back.loc != oldloc)
+		for(var/i = 0, i < max(get_dist(loc,back.loc),10),i++)
+			step_towards(back,oldloc)
+			if(loc == back.loc)
+				break
+		back.loc = oldloc//just in case
+
+	if(front && loc != front.oldloc)
+		for(var/i = 0, i < max(get_dist(loc,front.loc),10),i++)
+			step_towards(src,front.oldloc)
+			if(loc == front.loc)
+				break
+		loc = front.oldloc//just in case
+
+	oldloc = loc
+	gib_trail()
+	. = ..()
+
+/mob/living/simple_animal/hostile/eldritch/armsy/proc/gib_trail()
+	if(back)
+		return
+	var/chosen_decal = pick(typesof(/obj/effect/decal/cleanable/blood/gibs))
+	var/obj/effect/decal/cleanable/blood/gibs/decal = new chosen_decal(drop_location())
+	decal.setDir(dir)
+	return
+
+/mob/living/simple_animal/hostile/eldritch/armsy/Destroy()
+	if(front)
+		front.icon_state = "armsy_end"
+		front.icon_living = "armsy_end"
+	if(back)
+		back.Destroy() // chain destruction baby
+	. = ..()
+
+/mob/living/simple_animal/hostile/eldritch/armsy/proc/heal()
+	if(health == maxHealth)
+		if(back)
+			back.heal()
+			return
+		else
+			current_stacks++
+			if(current_stacks >= stacks_to_grow)
+				var/mob/living/simple_animal/hostile/eldritch/armsy/prev = new type(drop_location(),spawn_more = FALSE)
+				icon_state = "armsy_mid"
+				icon_living =  "armsy_mid"
+				back = prev
+				prev.icon_state = "armsy_end"
+				prev.icon_living = "armsy_end"
+				prev.front = src
+				prev.AIStatus = AI_OFF
+
+				current_stacks = 0
+
+	adjustBruteLoss(-maxHealth/2)
+	adjustFireLoss(-maxHealth/2)
+	adjustToxLoss(-maxHealth/2)
+	adjustOxyLoss(-maxHealth/2)
+
+/mob/living/simple_animal/hostile/eldritch/armsy/AttackingTarget()
+	if(istype(target,/obj/item/bodypart/r_arm) || istype(target,/obj/item/bodypart/l_arm))
+		target.Destroy()
+		heal()
+		return
+	if(back)
+		back.target = target
+		back.AttackingTarget()
+	if(!Adjacent(target))
+		return
+	do_attack_animation(target)
+	if(iscarbon(target))
+		var/mob/living/carbon/C = target
+		if(HAS_TRAIT(C, TRAIT_NODISMEMBER))
+			return
+		var/list/parts = list()
+		for(var/X in C.bodyparts)
+			var/obj/item/bodypart/BP = X
+			if(BP.body_part != HEAD && BP.body_part != CHEST && BP.body_part != LEG_LEFT && BP.body_part != LEG_RIGHT)
+				if(BP.dismemberable)
+					parts += BP
+		if(prob(10))
+			var/obj/item/bodypart/BP = pick(parts)
+			if(BP)
+				BP.dismember()
+
+	. = ..()
```

Comment:
```suggestion
	return ..()

```
Return at the end of the proc, instead of assigning the return to a variable that will be returned immediately.
If you are leaving empty lines inside the proc, leaving two between it and the next one helps with clarity as well.